### PR TITLE
Feature/add nightly

### LINKED ISSUE #735: [FEATURE] Nightly build
**Describe the feature request**

Add a Nightly build that runs a full clang-tidy compilation on all targets, including the benchmark, package tests, etc.

## Background
Due to compilation time, the following decisions were made in the past:

* In #345 , the clang-tidy build was restricted to the C API, and building clang-tidy on the tests was disabled from every PR and `main` build.
* During #718 , it was decided to remove the benchmark build from every PR and from every `main` build.

The downside of those decisions is, that there is the danger that the omitted parts of the compilation break and that that is unnoticed.

This feature will introduce a Nightly build that re-enables those checks

## Desired implementation

* [x] exclude build target `power_grid_model_benchmark_cpp` from the `ALL` target using `EXCLUDE_FROM_ALL`
* [x] add `all power_grid_model_benchmark_cpp` to the `options` for the `target` `inputs` for the `workflow_dispatch` in `.github/workflows/clang-tidy.yml`
  * [x] This is possible cfr. https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-build-t
* [x] enable `on` `workflow_call` for `.github/workflows/nightly.yml` and `.github/workflows/clang-tidy.yml` cfr. https://docs.github.com/en/actions/sharing-automations/reusing-workflows
* [x] add `.github/workflows/nightly.yml`
  * [x] call `.github/workflows/main.yml` and `.github/workflows/clang-tidy.yml` from this workflow cfr. https://docs.github.com/en/actions/sharing-automations/reusing-workflows

### PR DESCRIPTION
**Pull request by mgovers**
* [x] Adds support for nightly build
* [x] Disables benchmark from the `all` target

Closes #735

### CODE REVIEW

**Comment by TonyXiang8787** [comment]
Why are there so many code change? I thought this is only adding new workflow?

**Comment by mgovers** [comment]
> Why are there so many code change? I thought this is only adding new workflow?

to prove that the workflow does indeed call the correct parameters. I would prefer the first version of the workflow to pass, and then we can start there.

however, there's no target that currently passes in clang-tidy other than the `power_grid_model_c` target. This adds `power_grid_model_cpp` to `clang-tidy` as well.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ on:
   pull_request:
   # run pipeline on merge queue
   merge_group:
+  # run pipeline from another workflow
+  workflow_call:
```
Comment (TonyXiang8787) [inline .github/workflows/main.yml:18]: Why do we need to run `main.yml` in nightly build? We just need `clang-tidy` right?

Comment (mgovers) [reply .github/workflows/main.yml:18]: we don't need to but i'd consider it best practice.

in addition, there's no harm in enabling this option now. if you have strong feelings about it, i can remove it

Comment (TonyXiang8787) [reply .github/workflows/main.yml:18]: It's fine to also run main

**Review by figueroa1395 — APPROVED** [review]

**Pull request merged**